src/storybook/story-test-runner/extend-stories.ts

```diff
@@ -0,0 +1,115 @@
+import TypedModule from "module";
+import type { HermioneMetaConfig, HermioneStoryConfig } from "../../types";
+import type { StorybookStory, StorybookStoryExtended } from "./types";
+
+const Module = TypedModule as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+type ImportingModule = {
+    _compile: (code: string, fileName: string) => unknown;
+};
+
+type StoryFile = { default: HermioneMetaConfig } & Record<string, HermioneStoryConfig>;
+
+export function extendStoriesFromStoryFile(stories: StorybookStory[]): StorybookStoryExtended[] {
+    const storiesMap = getStoriesMap(stories);
+    const storyFile = getStoryFile(stories[0].absolutePath);
+    const withStoryFileExtendedStories = stories as StorybookStoryExtended[];
+
+    for (const storyName in storyFile) {
+        if (storyName === "default") {
+            withStoryFileExtendedStories.forEach(story => {
+                const hermioneStoryOpts = storyFile[storyName].hermione || {};
+
+                story.skip = hermioneStoryOpts.skip || false;
+                story.assertViewOpts = hermioneStoryOpts.assertViewOpts || {};
+                story.browserIds = hermioneStoryOpts.browserIds || null;
+            });
+
+            continue;
+        }
+
+        const storyMapKey = getStoryNameId(storyName);
+
+        if (storiesMap.has(storyMapKey) && storyFile[storyName].hermione) {
```

Comment:
Matching story export to story name by order of ASCII characters.

Its effectively like `_.toStartCase`, which [storybook is specified in docs](https://storybook.js.org/docs/api/csf#named-story-exports).